System/Console/Terminfo/Base.hs

```diff
@@ -93,7 +85,12 @@ setupTerm term = bracket (peek cur_term) (poke cur_term) $ \_ ->
             then throwIO $ SetupTermError
                 $ "Couldn't look up terminfo entry " ++ show term
             else do
-                cterm <- peek cur_term
+                -- There is no function to simply return the cur_term pointer.
+                -- We call set_curterm with nullPtr which will give us
+                -- the value of the old cur_term and then we set it back
+                -- again.
+                cterm <- set_curterm nullPtr
+                set_curterm cterm
```

Comment:
> There is no function to simply return the cur_term pointer.

Maybe simply reading a `cur_term` variable is that function? One reading should work better than two resettings.